### PR TITLE
AtariTerminalOnLifeLossWrapper

### PR DESCRIPTION
**Pull request by emailweixu**
Wrapper to change discount to 0 upon life loss for Atari.

This can potentially make it easier for the learning agent to recognize the
signficance of losing a life.

Some papers report the results with this enabled (e.g. arXiv:2111.00210).

### CODE REVIEW

File: alf/environments/alf_wrappers.py [hunk 1]
```diff
@@ -933,3 +933,34 @@ def _step(self, action):
...
+        self._prev_lives = torch.zeros(env.batch_size, dtype=torch.long)
+
+    def _reset(self):
+        time_step = self._env.reset()
+        self._prev_lives = time_step.env_info['ale.lives']
+        return time_step
+
+    def _step(self, action):
+        time_step = self._env.step(action)
+        lives = time_step.env_info['ale.lives']
```
Comment (hnyu) [inline alf/environments/alf_wrappers.py:964]: By default, 'ale.lives' might not exist in the env info?

File: alf/environments/alf_wrappers.py [hunk 2]
```diff
@@ -933,3 +933,34 @@ def _step(self, action):
...
+        time_step.discount[lives < self._prev_lives] = 0.
```
Comment (le-horizon) [inline alf/environments/alf_wrappers.py]: This line needs to be the following, otherwise it fails when there is only one env:
```
        if len(time_step.discount.shape) > 0:
            time_step.discount[lives < self._prev_lives] = np.zeros((), dtype=np.float32)
        elif lives < self._prev_lives:
            time_step = time_step._replace(discount=np.zeros((), dtype=np.float32))
```

File: alf/environments/alf_wrappers.py [hunk 3]
```diff
@@ -933,3 +933,34 @@ def _step(self, action):
         # action: [B, action_dim] or [action_dim]
         time_step = self._env.step(action)
         return time_step._replace(prev_action=prev_action)
+
+
```
Comment (le-horizon) [inline alf/environments/alf_wrappers.py:937]: Would be good to have
`@alf.configurable` so gin config can also use this class.

File: alf/environments/alf_wrappers.py (hunk 1 above)
Comment (emailweixu) [reply alf/environments/alf_wrappers.py:964]: gym.envs.atari.atari_env always output "ale.lives".

File: alf/environments/alf_wrappers.py (hunk 2 above)
Comment (emailweixu) [reply alf/environments/alf_wrappers.py]: Previously, this wrapper is intended to be used as create_environment.batched_wrappers. Now I changed it to use as suite_gym.load.alf_env_wrappers.

File: alf/environments/alf_wrappers.py (hunk 3 above)
Comment (emailweixu) [reply alf/environments/alf_wrappers.py:937]: added

**Review by le-horizon — APPROVED** [review]
LG
